### DISCUSSION TITLE
New project setup: export OPENLANE_ROOT does not work

### DISCUSSION THREAD

**Discussion by srinivasf1** [efabless/caravel_user_project#256] [Q&A] [answered]
I'm trying to setup a new project and following documentation from here : https://caravel-user-project.readthedocs.io/en/latest/#overview

I'm in my project directory. I run the export command for OPENLANE_ROOT and PDK_ROOT. However, after I run make setup, I get the following error

OPENLANE_ROOT is set to '/Users/mymac/Silicon_exp1/openlane' which contains openlane config files
Please set it to a different directory
make: *** [openlane] Error 1

Is this the export command that's failing? Is there a workaround?

**Comment by vijayank88** [comment]
Don't point existing `OpenLane` installed directory.
Caravel flow works on specific tag, so provide separate path for `OPENLANE_ROOT` and `PDK_ROOT`
Preferably with caravel flow directory itself create as per instruction in the guide.

**Reply by srinivasf1** [reply]
In my project directory there's an OpenLane directory. I have created another directory called dependencies and I set OPENLAN_ROOT to that path
export OPENLANE_ROOT=$(pwd)/dependencies/openlane_src

I'm following the instruction from here https://caravel-user-project.readthedocs.io/en/latest/

Isn't this the way?

**Reply by srinivasf1** [reply]
I tried creating dependencies folder outside of my project folder. I ran the export command and again, the variable is being set to the directory. I know this because that is what I see when I check the variable.

`echo $OPENLANE_ROOT'
'/Users/$user/Documents/Silicon_projects/dependencies/openlane_src`


I'm on Mac OS if that matters.

**Comment by srinivasf1** [comment, marked as answer]
Figured out the reason.

In Makefile, the env variable for open lane is verified using 

`.PHONY: openlane
openlane:
        @if [ "$$(realpath $${OPENLANE_ROOT})" = "$$(realpath $$(pwd)/openlane)" ]; then\
                echo "OPENLANE_ROOT is set to '$$(pwd)/openlane' which contains openlane config files"; \
                echo "Please set it to a different directory"; \
                exit 1; \
        fi
        cd openlane && $(MAKE) openlane
`

Mac does not have realpath. The setup works after installing realpath via homebrew.

Here's stackoverflow solution to getting realpath : https://stackoverflow.com/questions/3572030/bash-script-absolute-path-with-os-x
